### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by AurevoirXavier**
- remove `init.sh` which will setup a wrong toolchain version
- update `README.md`:
	- add: **Crayfish Testnet** intro; **[darwinia-builder](https://github.com/AurevoirXavier/darwinia-builder)** for env setup and building
	- fix: image link, env setup step
- add `.rustfmt.toml` for **rustfmt**, follow Substrate coding style
- remove all unused **import**
- remove all unnecessary `clone`
- follow `cargo clippy` suggestions
- ignore *cross compile files* and *cargo expand files*
- refactor **staking** module
- rename `evo-staking` to `darwinia-staking`
- add tests for **staking** module

### CODE REVIEW

File: srml/staking/src/lib.rs [hunk 1]
```diff
@@ -985,57 +954,61 @@ impl<T: Trait> Module<T> {
...
-								if !item.value.is_zero() {
-									Some(item)
-								} else {
-									None
-								}
-							}
-						})
-						.collect::<Vec<_>>();
-					ledger.deposit_items = new_deposit_items;
+					deposit_items.drain_filter(|item| {
```
Comment (HackFisher) [inline srml/staking/src/lib.rs]: 看一下no_std 是否支持drain_filter

**Comment by HackFisher** [comment]
.rustfmt.toml is missing, may need to add back.

**Comment by HackFisher** [comment]
https://doc.rust-lang.org/1.37.0/alloc/vec/struct.DrainFilter.html

File: srml/staking/src/lib.rs (hunk 1 above)
Comment (AurevoirXavier) [reply srml/staking/src/lib.rs]: [drain_filter](https://doc.rust-lang.org/alloc/vec/struct.Vec.html#method.drain_filter) can pass the wasm compiling
use `feature` cuz the uncertain name [discuss for drain_filter](https://github.com/rust-lang/rust/issues/43244)

check `no_std` [playground](https://play.rust-lang.org/?version=nightly&mode=release&edition=2018&gist=143b6fa3b348aa00ce0599647518630e)

Comment (AurevoirXavier) [reply srml/staking/src/lib.rs]: also used in libra [ref](https://github.com/libra/libra/issues/270)

**Pull request merged**